### PR TITLE
ADYEN-585 setup adyen v2/v3 package and move adyen v2/v3 payment strategy

### PR DESCRIPTION
**Pull request by vitalii-koshovyi**
## What?
Setup adyen v2/v3 package and move adyen v2/v3 payment strategy

## Why?
As part of monorepo setup move adyen v2/v3 to a separate package.

## Testing / Proof
circle/manually

@bigcommerce/checkout @bigcommerce/payments

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -17098,7 +17098,7 @@
     "filter-obj": {
       "version": "1.1.0",
       "resolved": "https://registry.npmjs.org/filter-obj/-/filter-obj-1.1.0.tgz",
-      "integrity": "sha512-8rXg1ZnX7xzy2NGDVkBVaAy+lSlPNwad13BtgSlLuxfIslyt5Vg64U7tFcCt4WS1R0hvtnQybT/IyCkGZ3DpXQ=="
+      "integrity": "sha1-mzERErxsYSehbgFsbF1/GeCAXFs="
```
Comment (bc-peng) [inline package-lock.json:17101]: Is this intentional? Thought It was strange to only change this one line.

**Review by bc-peng — COMMENTED** [review]
This PR touched so many files. Could you please separate adyen code from the changes to the `payment-integration-api` and make another PR for the non-adyen spcific content?

Also, can we rename the PR title by following the style on https://github.com/bigcommerce/checkout-sdk-js/pulls? 
If there is no respectie Jira ticket number, you may use `CHCKOUT-0000`.

File: package-lock.json (hunk 1 above)
Comment (vitalii-koshovyi) [reply package-lock.json:17101]: oh no, it's not

Comment (animesh1987) [reply package-lock.json:17101]: Can we revert this?

File: packages/adyen-integration/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    preset: "../../jest.preset.js",
+    globals: {
+        "ts-jest": {
+            tsconfig: "adyen-integration/tsconfig.spec.json",
+            diagnostics: false,
+        },
+    },
+    setupFilesAfterEnv: ["../../jest-setup.js"],
+    coverageDirectory: "../../coverage/packages/adyen-integration",
+};
```
Comment (animesh1987) [inline packages/adyen-integration/jest.config.js:12]: EOL

File: packages/payment-integration-api/src/payment/payments.mock.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { CreditCardInstrument } from './payment';
+
+
+export function getCreditCardInstrument(): CreditCardInstrument {
```
Comment (animesh1987) [inline packages/payment-integration-api/src/payment/payments.mock.ts:4]: Should this be in test Utils?

File: packages/payment-integration-api/src/payment/payment-actions.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { Action } from '@bigcommerce/data-store';
+
+import { LoadOrderAction } from '../order';
+
+import PaymentResponseBody from './payment-response-body';
+
+export enum PaymentActionType {
```
Comment (animesh1987) [inline packages/payment-integration-api/src/payment/payment-actions.ts:7]: Do we need to move this across?

File: packages/core/src/payment/strategies/adyenv2/index.ts [hunk 5]
```diff
@@ -1,5 +1,3 @@
 export * from './adyenv2';
 
-export { default as AdyenV2ScriptLoader } from './adyenv2-script-loader';
-export { default as AdyenV2PaymentStrategy } from './adyenv2-payment-strategy';
-export { default as AdyenV2PaymentInitializeOptions } from './adyenv2-initialize-options';
+export { default as AdyenV2ScriptLoader } from './adyenv2-script-loader';
```
Comment (animesh1987) [inline packages/core/src/payment/strategies/adyenv2/index.ts:3]: EOL

Comment (animesh1987) [reply packages/core/src/payment/strategies/adyenv2/index.ts:3]: Are we going to move the script loader across as well?

File: packages/adyen-integration/src/adyenv2/adyenv2-method.mock.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+            testMode: true,
+        },
+        initializationData: {
+            originKey: 'YOUR_ORIGIN_KEY',
+            clientKey: 'YOUR_CLIENT_KEY',
+        },
+        type: 'PAYMENT_TYPE_API',
+        clientToken: 'clientToken',
+    };
+}
```
Comment (animesh1987) [inline packages/adyen-integration/src/adyenv2/adyenv2-method.mock.ts:21]: EOL

File: packages/adyen-integration/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
+    "extends": "../../tsconfig.base.json"
+}
```
Comment (animesh1987) [inline packages/adyen-integration/tsconfig.json:3]: EOL

File: package-lock.json (hunk 1 above)
Comment (vitalii-koshovyi) [reply package-lock.json:17101]: For sure, I'll do it when I receive all review comments

File: packages/adyen-integration/src/adyenv3/adyenv3-method.mock.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+            testMode: true,
+        },
+        initializationData: {
+            originKey: 'YOUR_ORIGIN_KEY',
+            clientKey: 'YOUR_CLIENT_KEY',
+        },
+        type: 'PAYMENT_TYPE_API',
+        clientToken: 'clientToken',
+    };
+}
```
Comment (animesh1987) [inline packages/adyen-integration/src/adyenv3/adyenv3-method.mock.ts:21]: EOL

File: packages/payment-integration-api/src/payment/payment-actions.ts (hunk 4 above)
Comment (vitalii-koshovyi) [reply packages/payment-integration-api/src/payment/payment-actions.ts:7]: yes, we use it in the SubmitPaymentAction

File: packages/payment-integration-api/src/address/index.ts [hunk 9]
```diff
@@ -1 +1,2 @@
 export { default as Address, AddressKey, AddressRequestBody } from "./address";
+export { default as InternalAddress } from "./internal-address";
```
Comment (animesh1987) [inline packages/payment-integration-api/src/address/index.ts:2]: Do we still need this? In theory we should be deleting this folder completely?

File: packages/core/src/payment/strategies/adyenv2/index.ts (hunk 5 above)
Comment (animesh1987) [reply packages/core/src/payment/strategies/adyenv2/index.ts:3]: From what I remember this will stay due to Google payment strategies rely on this?

Comment (vitalii-koshovyi) [reply packages/core/src/payment/strategies/adyenv2/index.ts:3]: yes

File: packages/payment-integration-api/src/address/index.ts (hunk 9 above)
Comment (vitalii-koshovyi) [reply packages/payment-integration-api/src/address/index.ts:2]: It uses in chain by SubmitPaymentAction

File: packages/adyen-integration/tsconfig.json (hunk 7 above)
Comment (animesh1987) [inline packages/adyen-integration/tsconfig.json:3]: EOL

File: packages/core/src/payment/strategies/adyenv2/index.ts (hunk 5 above)
Comment (animesh1987) [inline packages/core/src/payment/strategies/adyenv2/index.ts:3]: EOL

**Review by animesh1987 — APPROVED** [review]
Nice work @vitalii-koshovyi

**Pull request merged**